### PR TITLE
FEATURE: Allow turning off the plugin from settings

### PR DESCRIPTION
**Pull request by xfalcox**

### CODE REVIEW

File: assets/javascripts/discourse/connectors/topic-footer-main-buttons-before-create/assign-button.js.es6 [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { ajax } from 'discourse/lib/ajax';
 
 export default {
   shouldRender(args, component) {
-    const needsButton = component.currentUser && component.currentUser.get('staff');
+    const needsButton = component.currentUser && component.currentUser.get('staff') && component.siteSettings.assign_enabled;
```
Comment (eviltrout) [inline assets/javascripts/discourse/connectors/topic-footer-main-buttons-before-create/assign-button.js.es6]: If the plugin is disabled, connectors are not supposed to be included. Did you find that they were?

https://github.com/discourse/discourse/blob/master/app/assets/javascripts/discourse/lib/plugin-connectors.js.es6#L29

Comment (xfalcox) [inline assets/javascripts/discourse/connectors/topic-footer-main-buttons-before-create/assign-button.js.es6]: I added the `enabled_site_setting :assign_enabled` to the top of the plugin.rb file but the buttons were still there in many places.

Comment (eviltrout) [inline assets/javascripts/discourse/connectors/topic-footer-main-buttons-before-create/assign-button.js.es6]: I would love to know why! Did it not come through in `Discourse.Site.currentProp('disabled_plugins')`?

Comment (xfalcox) [inline assets/javascripts/discourse/connectors/topic-footer-main-buttons-before-create/assign-button.js.es6]: It is inside the `disabled_plugins`, but the button on topic end still appears:

image

Comment (eviltrout) [inline assets/javascripts/discourse/connectors/topic-footer-main-buttons-before-create/assign-button.js.es6]: Ah I know why! It needs the path to include `discourse-assign` which it does not with this configuration.

Try moving `assets/javascripts/discourse` within the plugin to be `assets/javascripts/discourse-assign` - then the matcher will work.

Comment (xfalcox) [inline assets/javascripts/discourse/connectors/topic-footer-main-buttons-before-create/assign-button.js.es6]: Wow, this solves 90% of the problem.

Still:

- `api.addUserMenuGlyph`
- `api.addTagsHtmlCallback`
- `api.decorateWidget('post-contents:after-cooked')`

run and add stuff to the interface.

Comment (eviltrout) [inline assets/javascripts/discourse/connectors/topic-footer-main-buttons-before-create/assign-button.js.es6]: Would be great to get to those too eventually. I just thought you could remove the connector specific stuff for this PR :)

Comment (xfalcox) [inline assets/javascripts/discourse/connectors/topic-footer-main-buttons-before-create/assign-button.js.es6]: Done, thanks @eviltrout !

**Pull request merged**